repair/row_level.cc

```diff
@@ -2836,9 +2837,18 @@ future<> repair_service::init_ms_handlers() {
             auto& table = local_repair.get_db().local().find_column_family(gid.table);
             auto erm = table.get_effective_replication_map();
             auto& tmap = erm->get_token_metadata_ptr()->tablets().get_tablet_map(gid.table);
+            auto* trinfo = tmap.get_tablet_transition_info(gid.tablet);
+            if (!trinfo) {
+                rlogger.warn("Skipped repair_update_compaction_ctrl gid={} session_id={} since tablet is not in transition", gid, topo_guard);
+                co_return;
+            }
+            if (trinfo->stage != locator::tablet_transition_stage::end_repair) {
+                rlogger.warn("Skipped repair_update_compaction_ctrl gid={} session_id={} since tablet is not in tablet_transition_stage::end_repair", gid, topo_guard);
+                co_return;
```

Comment:
We should fail the RPC in case of those exits, so that the coordinator doesn't think it was successful.